packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -501,21 +501,29 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin
   @override
   Widget build(BuildContext context) {
     assert(position != null);
-    // TODO(ianh): Having all these global keys is sad.
-    Widget result = RawGestureDetector(
-      key: _gestureDetectorKey,
-      gestures: _gestureRecognizers,
-      behavior: HitTestBehavior.opaque,
-      excludeFromSemantics: widget.excludeFromSemantics,
-      child: Semantics(
-        explicitChildNodes: !widget.excludeFromSemantics,
-        child: IgnorePointer(
-          key: _ignorePointerKey,
-          ignoring: _shouldIgnorePointer,
-          ignoringSemantics: false,
-          child: _ScrollableScope(
-            scrollable: this,
-            position: position,
+    /// [_ScrollableScope] must be placed above the [BuildContext] returned by [notificationContext]
+    /// so that we can get this [ScrollableState] by doing the following:
+    /// ```dart
+    /// ScrollNotification notification;
+    /// Scrollable.of(notification.context)
+    /// ```
+    /// Since [notificationContext] is pointing to [_gestureDetectorKey.context], [_ScrollableScope]
+    /// must be placed above the widget using it: [RawGestureDetector]
```

Comment:
thanks, this is much clearer.

nit: these should just be regular `//` comments since they're not actually documentation (as in, they don't end up in the api docs pages)

Reply:
Documentation comments on private classes won't appear in the API docs, so it's not necessary, although having the comments is helpful.

Reply:
> Documentation comments on private classes won't appear in the API docs,

True but the flutter guidelines recommended it when possible, as it's not entirely impossible for private classes to become public later.

Reply:
Yes quite true.  Like I said, it's helpful, but not necessary.